lib/util/uint8array_utils.js

```diff
@@ -131,11 +131,49 @@ shaka.util.Uint8ArrayUtils.equal = function(arr1, arr2) {
 
 
 /**
- * Convert a Uint8Array to a string which can be used as a key in a dictionary.
+ * Generates a random unique number to use as a key for a given Unit8Array
+ * or returns the key for a a given Unit8Array
  * @param {!Uint8Array} arr
  * @return {string}
  */
-shaka.util.Uint8ArrayUtils.key = function(arr) {
-  return Array.prototype.join.apply(arr);
-};
-
+shaka.util.Uint8ArrayUtils.key = (function() {
+
+  var cache = {};
```

Comment:
I'm working on a patch right now to remove the Uint8Array key concept completely.  It will mean a theoretically slower check to suppress duplicate init datas (order N instead of order 1), but there shouldn't be so many unique init datas in a stream that it will be a problem in practice.

I'll update here again once that is in, and then you can drop this part of your patch and rework your dedup in your MediaKeys polyfill to match.

Sound reasonable?